upgrade/upgrade.go

```diff
@@ -231,6 +243,75 @@ func ConfigureDollarModule(ctx sdk.Context, dollarKeeper *dollarkeeper.Keeper) (
 	}
 }
 
+// ConfigureSwapModule creates an initial USDC/USDN swap pool.
+func ConfigureSwapModule(ctx sdk.Context, swapKeeper *swapkeeper.Keeper, accountKeeper authkeeper.AccountKeeper) (err error) {
+	switch ctx.ChainID() {
+	case MainnetChainID:
+		// Create the initial uusdn<>uusdc pool, following the same logic of the StableSwap `CreatePool` function:
+		// https://github.com/noble-assets/swap/blob/f94f41da984bdfbdebb013f70ed8ce05d2993726/keeper/msg_stableswap_server.go#L46-L169
+
+		// Increase and get the next Pool ID.
+		poolId, err := swapKeeper.IncreaseNextPoolID(ctx)
+		if err != nil {
+			return errors.Wrapf(err, "unable to set next pool id")
+		}
+
+		// Create the Pool address.
+		prefix := fmt.Sprintf("%s/pool/%d", swaptypes.ModuleName, poolId)
+		account := authtypes.NewEmptyModuleAccount(prefix)
+		account = accountKeeper.NewAccount(ctx, account).(*authtypes.ModuleAccount)
+		accountKeeper.SetModuleAccount(ctx, account)
+
+		// Create the Protocol Fees Pool address.
+		protocolFeesAccount := authtypes.NewEmptyModuleAccount(fmt.Sprintf("%s/protocol_fees", prefix))
+		protocolFees := accountKeeper.NewAccount(ctx, protocolFeesAccount).(*authtypes.ModuleAccount)
+		accountKeeper.SetModuleAccount(ctx, protocolFees)
+
+		// Create the Rewards Fees Pool address.
+		rewardFeesAccount := authtypes.NewEmptyModuleAccount(fmt.Sprintf("%s/reward_fees", prefix))
+		rewardFees := accountKeeper.NewAccount(ctx, rewardFeesAccount).(*authtypes.ModuleAccount)
+		accountKeeper.SetModuleAccount(ctx, rewardFees)
+
+		// Set the new Pool on state.
+		err = swapKeeper.SetPool(ctx, 0, swaptypes.Pool{
+			Id:        poolId,
+			Address:   account.GetAddress().String(),
+			Algorithm: swaptypes.STABLESWAP,
+			Pair:      "uusdc",
```

Comment:
Definitely overkill, but should we fetch the `uusdc` value from the FiatTokenFactory keeper?